src/main/kotlin/org/rust/lang/core/psi/RsPsiFactory.kt

```diff
@@ -368,8 +371,10 @@ private fun RsFunction.getSignatureText(subst: Substitution): String? {
 
 private fun String.iff(cond: Boolean) = if (cond) this + " " else " "
 
-private fun RsTypeReference.substAndGetText(subst: Substitution): String =
-    type.substitute(subst).insertionSafeTextWithLifetimes
+private fun RsTypeReference.substAndGetText(subst: Substitution): String {
+    val substitutedType = type.substitute(subst)
+    return if (this.refLikeType?.lifetime != null) substitutedType.insertionSafeTextWithLifetimes else substitutedType.insertionSafeText
```

Comment:
This line is too long, consider introducing `val hasLifetime = refLikeType?.lifetime != null`
Although there are many too long lines in the project, we try to avoid new ones 😄

Reply:
Sure thing, done.